features/0351-generic-json-message/README.md

```diff
@@ -0,0 +1,130 @@
+
+##### `data-type`
+
+The sender must provide a `data-type` identification string constant, which indicates, for the recipient, the format of the data and how it is to be processed. 
+
+##### `data`, `data~attach`, `~attach`
+
+These 3 fields are optional, although at least one must be present. They carry attachments (inline, embedded, and appended, respectively).
+
+The payload transported in this message is placed in an [attachment](/concepts/0017-attachments/README.md). If the attachment is embedded or appended, the recipient is provided with the entire attachment descriptor.
+
+The recipient-agent must be able to process all three fields.
+
+#### Threading
+
+A protocol built on top of generic JSON messages can include [`~thread` decorators](/concepts/0008-message-id-and-threading/README.md) in the messages. For example, this allows the recipient-agent to relay a reply from the recipient, so the recipient is able to reply to a message.
+
+For threading with senders or recipients that aren't aries agents, the sender-agent or recipient-agent must maintain context to correlate the DIDComm message thread, and the message thread of the communication protocol with the non-aries communicator. The lifetime of this context, for a custom protocol built on top of generic-JSON-message, is defined by the custom protocol.
+
+#### Non-DIDComm Message Formats
+
+The message sent by the sender to the sender-agent and the message sent by the recipient-agent to the recipient are expected to have their own formats defined within a specific protocol built on top of this RFC. Some general principles can guide their implementation and translation to/from the DIDComm messages sent from sender-agent to recipient-agent.
+
+- Headers and metadata can sometimes be translated into DIDComm decorators - HTTP keep-alive to DIDComm `~timing`, for example.
+- The messages in your protocol may have a message-type identifier, which could be used for the `data-type` value.
+
+## Drawbacks
+
+## Rationale and alternatives
+
+- Why don't we standardize data-type labels against content schemas and format definitions? The intent here is to allow custom protocols and messaging schemes to operate over DIDComm. For this we provide a message format that allows them to be constructed without a chance of collision (between message names, for example) with other DIDComm message types, and we allow custom formats to be designed without needing to standardize them through the RFC process. The data-type label allows implementors to ensure that they will not collide between any of their message types in their specific implementations. A custom schema could be defined in its own RFC.
+
+## Prior art
+
```

Comment:
DIF Identity Hubs have Actions. https://github.com/decentralized-identity/identity-hub/blob/master/explainer.md#actions

However, this RFC is describing an agent-based mechanism rather than a Hub one.